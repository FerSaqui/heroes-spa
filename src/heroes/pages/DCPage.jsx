import React from 'react'
import { Navbar } from '../../ui/components/Navbar'
import { HeroList } from '../components/HeroList';

export const DCPage = () => {
  return (
    <>
      <h1>DC Comics</h1>
      <hr />

      <HeroList publisher={"DC Comics"}/>
    </>
  )
};