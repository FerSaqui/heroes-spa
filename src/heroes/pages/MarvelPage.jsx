import React from 'react'
import { HeroList } from '../components/HeroList';

export const MarvelPage = () => {
  return (
    <>
      <h1>Margel Comics</h1>
      <hr />

      <HeroList publisher={"Marvel Comics"}/>
    </>
  )
};