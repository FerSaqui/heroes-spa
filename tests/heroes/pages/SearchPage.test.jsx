import { screen, render } from '@testing-library/react'
import { SearchPage } from '../../../src/heroes/pages/SearchPage';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { AuthContext } from '../../../src/auth/context/AuthContext';

jest.mock('query-string', () => ({
    parse: jest.fn(() => ({ q: 'batman' })), //Especificar los query params
    stringify: jest.fn()
    // Add other methods you need to mock here
}));

describe('pruebas en <SearchPage />', () => {
    const contextValue = {
        logged: true,
        user: {
            id: 1,
            name: 'Juan Carlos',
        },
        logout: jest.fn(),
    };

    const routesConfig = [
        {
            path: '/search',
            element: <SearchPage />,
        }
    ];

    const router = createMemoryRouter(routesConfig, {
        initialEntries: ['/search'],
    });
     
    beforeEach(() => jest.clearAllMocks());

    test('Debe de mostrarse correctamente con valores por defecto', () => {
        const { container } = render(
            <AuthContext.Provider value={contextValue}>
                <RouterProvider router={router} />
            </AuthContext.Provider>
        );

        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar Batman y el input con el valor del queryString', () => {
        render(
            <AuthContext.Provider value={contextValue}>
                <RouterProvider router={router} />
            </AuthContext.Provider>
        );

        const inputValue = screen.getByRole("textbox");
        expect(inputValue.value).toBe("batman");

        const img = screen.getByRole("img");
        expect(img.src).toContain("/assets/heroes/dc-batman.jpg");
    });
});