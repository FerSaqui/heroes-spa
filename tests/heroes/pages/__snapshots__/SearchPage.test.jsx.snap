// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <SearchPage /> Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value="batman"
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
        data-testid="divSearchHeroeTxt"
        style="display: none;"
      >
        Search a Hero
      </div>
      <div
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No hero with 
        <b>
          batman
        </b>
      </div>
      <div
        class="col animate__animated animate__fadeIn"
      >
        <div
          class="card"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-4"
            >
              <img
                alt="Batman"
                class="card-img"
                src="/assets/heroes/dc-batman.jpg"
              />
            </div>
            <div
              class="col-8"
            >
              <div
                class="card-body"
              >
                <h5
                  class="card-title"
                >
                  Batman
                </h5>
                <p
                  class="card-text"
                >
                  Bruce Wayne
                </p>
                <p
                  class="card-text"
                >
                  <small
                    class="text-muted"
                  >
                    Detective Comics #27
                  </small>
                </p>
                <a
                  href="/hero/dc-batman"
                >
                  Más información
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
